node-tests/acceptance/build-test.js

```diff
@@ -1,31 +1,349 @@
+      let app = new AddonTestApp();
+      let engineName = 'eager';
+      let addonName = 'nested';
+
+      yield app.create('engine-testing', { noFixtures: true });
+      let engine = yield InRepoEngine.generate(app, engineName, { lazy: false });
+      let addon = yield engine.generateNestedAddon(addonName);
+
+      addon.writeFixture({
+        app: {
+          'foo.js': '// should wind up in engine namespace'
+        },
+        addon: {
+          components: {
+            'bar.js': '// should wind up in addon namespace'
+          },
+          templates: {
+            components: {
+              'bar.hbs': '<h1>bar</h1>'
+            }
+          }
+        }
+      });
+
+      let output = yield build(app);
+
+      // Verify app files in addon wind up in engine's namespace in vendor.js
+      output.contains('assets/vendor.js', moduleMatcher(`${engineName}/foo`));
+
+      // Verify addon files in addon wind up in addon's namespace in vendor.js
+      output.contains('assets/vendor.js', moduleMatcher(`${addonName}/components/bar`));
+      output.contains('assets/vendor.js', moduleMatcher(`${addonName}/templates/components/bar`));
+    }));
+
+    it('correctly builds addon in lazy engine', co.wrap(function* () {
+      let app = new AddonTestApp();
+      let engineName = 'lazy';
+      let addonName = 'nested';
+
+      yield app.create('engine-testing', { noFixtures: true });
+      let engine = yield InRepoEngine.generate(app, engineName, { lazy: true });
+      let addon = yield engine.generateNestedAddon(addonName);
+
+      addon.writeFixture({
+        app: {
+          'foo.js': '// should wind up in engine namespace'
+        },
+        addon: {
+          components: {
+            'bar.js': '// should wind up in addon namespace'
+          },
+          templates: {
+            components: {
+              'bar.hbs': '<h1>bar</h1>'
+            }
+          }
+        }
+      });
+
+      let output = yield build(app);
+
+      // Verify app files in addon wind up in engine's namespace in engine.js
+      output.contains(`engines-dist/${engineName}/assets/engine.js`, moduleMatcher(`${engineName}/foo`));
+
+      // Verify addon files in addon wind up in addon's namespace in enigne-vendor.js
+      output.contains(`engines-dist/${engineName}/assets/engine-vendor.js`, moduleMatcher(`${addonName}/components/bar`));
+      output.contains(`engines-dist/${engineName}/assets/engine-vendor.js`, moduleMatcher(`${addonName}/templates/components/bar`));
+    }));
+
+    it('invokes trees and executes hooks in the proper order and handles imports for engines', co.wrap(function* () {
+      let app = new AddonTestApp();
+      let engineName = 'tree-invocation-order';
+
+      yield app.create('engine-testing', { noFixtures: true });
+      let engine = yield InRepoEngine.generate(app, engineName, { lazy: true });
+
+      let fixture = fixturify.readSync(path.resolve(__dirname, '../fixtures/tree-invocation-order'));
```

Comment:
ya, makes sense.  And we still may want separate file for [index.js](https://github.com/trentmwillis/ember-engines/blob/173c5969d772817adef61ca508a81b41cba833a4/node-tests/fixtures/tree-invocation-order/index.js), but most of the other files would easily be inlined.

Totally up to you on inlining all or just the "smallish" ones...